modules/database/repository.py

```
@@ -0,0 +1,111 @@
+        try:
+            database_entry: WebhookDatabaseModel = self._database.add_webhook(webhook_type=webhook.trigger)
+            data: RecentlyAddedWebhookData = webhook.data  # type: ignore
+            library_name = data.library_name
+            item_name = data.title
+            _ = self.add_recently_added_item_to_database(webhook=database_entry,
+                                                         library_name=library_name,
+                                                         item_name=item_name)
+
+            return WebhookModel.from_database_record(record=database_entry)
+        except Exception as e:
+            raise Exception(f'Error adding webhook to database: {e}')
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)